bim2sim/elements/mapping/ifc2python.py

```diff
@@ -275,8 +275,10 @@ def get_property_set_by_name(property_set_name: str, element: entity_instance,
     """
     property_dict = None
     all_property_sets_list = element.IsDefinedBy
+    preselection = [item for item in all_property_sets_list
+        if hasattr(item, 'RelatingPropertyDefinition')]
     property_set = next(
-        (item for item in all_property_sets_list if
+        (item for item in preselection if
          item.RelatingPropertyDefinition.Name == property_set_name), None)
     if hasattr(property_set, 'RelatingPropertyDefinition'):
```

Comment:
Why do you do a preselection of the property sets by checking `hasattr(item, 'RelatingPropertyDefinition')` and then do again in if clause `if hasattr(property_set, 'RelatingPropertyDefinition'):` in the next step? Don't get the changes you did @veronikarichter

Reply:
@DaJansenGit 
We use a preselection to include all items that have the item 'RelatingPropertyDefinition'. In the next step, we define the property_set by iterating through the preselection, where we add all items to the property_set that match the property_set_name. If the item.RelatingPropertyDefinition.Name does not match the property_set_name, the item is set to None. 
By having the additional check in Line 283, we avoid running in an item that is set to None. At least this was an issue that occurred during debugging for me.